File: 767-prime-number-of-set-bits-in-binary-representation/prime-number-of-set-bits-in-binary-representation.py
class Solution:
    def countPrimeSetBits(self, left: int, right: int) -> int:
        # A bitmask where the 2nd, 3rd, 5th, 7th, 11th, 13th, 17th, and 19th bits are set
        # Binary: 10100010100010101100
        prime_mask = 665772
        count = 0
        
        for num in range(left, right + 1):
            # bin(num).count('1') is the Pythonic way to count set bits
            bits = bin(num).count('1')
            
            # Check if the 'bits'-th position in our mask is a 1
            if (prime_mask >> bits) & 1:
                count += 1
                
        return count

File: 767-prime-number-of-set-bits-in-binary-representation/prime-number-of-set-bits-in-binary-representation.java
class Solution {
    public int countPrimeSetBits(int left, int right) {

        HashSet<Integer> prime = new HashSet<Integer>();
        prime.add(2);
        prime.add(3);
        prime.add(5);
        prime.add(7);
        prime.add(11);
        prime.add(13);
        prime.add(17);
        prime.add(19);

        int count = 0;
        while (left <= right) {
            int i = left;
            int bits = 0;
            while (i > 0) {
                if (i % 2 == 1) {
                    bits++;
                }
                i = i / 2;
            }
            if (prime.contains(bits)) {
                count++;
            }
            left++;
        }
        return count;
    }
}